### PR TITLE
Set State.WaitingForInput in PrepareForInput

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/interactive

File: src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs [hunk 1]
```diff
@@ -719,8 +719,8 @@ private void AppendInput(string text)
             public void PrepareForInput()
             {
                 _buffer.Flush();
-
                 AddLanguageBuffer();
+                State = State.WaitingForInput;
```
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs]: As suggested by @amcasey.

File: src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs [hunk 2]
```diff
@@ -1506,6 +1506,32 @@ public void CutLineWithSelection()
...
+        public void SubmitAsyncMultiple()
+        {
+            SubmitAsync("1", "2", "1 + 2");
+        }
+
+        private void SubmitAsync(params string[] submissions)
+        {
+            var actualSubmissions = new List<string>();
+            var evaluator = _testHost.Evaluator;
+            EventHandler<string> onExecute = (_, s) => actualSubmissions.Add(s.TrimEnd());
```
Comment (genlu) [inline src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs]: What is the `TrimEnd` for? New line?

**Comment by genlu** [comment]
:+1:

**Comment by amcasey** [comment]
:+1:

File: src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs (hunk 2 above)
Comment (cston) [inline src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs]: Yes, to trim the new line.

**Comment by MattGertz** [comment]
Approved pending successful test completion.

**Comment by ManishJayaswal** [comment]
@cston this needs to be merged into stabilization branch. Master in not going into 1.1 release.

LGTM otherwise.

**Comment by cston** [comment]
@ManishJayaswal Good catch, thanks. The PR should have been from stabilization originally. Ported in #6176.

**Pull request merged**